evidence.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package psatoken
+
+import (
+	"crypto"
+	"crypto/rand"
+	_ "crypto/sha256" // used hash algorithms need to be imported explicitly
+	"errors"
+	"fmt"
+
+	cose "github.com/veraison/go-cose"
+)
+
+// Evidence is the wrapper around the PSA token, including the COSE envelope and
+// the underlying claims
+// nolint: golint
+type Evidence struct {
+	Claims  Claims
+	message *cose.Sign1Message
+}
+
+// SetClaims attaches the supplied claims to the Evidence instance. The claims
+// are checked for consistency with the given profile.
+func (e *Evidence) SetClaims(claims *Claims, profile string) error {
+	if err := checkSupportedProfiles(profile); err != nil {
+		return err
+	}
+
+	err := claims.validate(profile)
```

Comment:
Little surprised, line 31, err is set as a new variable and compiler is not complaining?

Reply:
the lifetime of `err` at L27 is scoped to the conditional statement only.